charts/observability-pipelines-worker/templates/configmap.yaml

```diff
@@ -8,7 +8,7 @@ metadata:
     checksum/config: {{ tpl (toYaml .Values.config) . | sha256sum }}
 data:
   {{- if .Values.config }}
-  config.yaml: | {{ tpl (toYaml .Values.config) . | nindent 4 }}
+  pipelines.yaml: | {{ tpl (toYaml .Values.config) . | nindent 4 }}
```

Comment:
I'd like to get more familiar with the wiring up of the config. Is the following correct?

- Users must provide a config value
- The config is turned into a `ConfigMap` (i.e. here)
- The `ConfigMap` is made available in a volume at `/etc/observability-pipelines-worker/`
- OPW can then read from `/etc/observability-pipelines-worker/pipelines.yaml`

And if users already have their own config file at a different location, are they meant to override the `args` to use it?

Reply:
1. User provides contents to the `config` key in the `values.yaml`.
2. `configmap.yaml` will take those contents try to parse them as YAML.
3. This [volume](https://github.com/DataDog/helm-charts/blob/0e9cc065f09c94cdc54ccdfc3e2f66a9e22620d7/charts/observability-pipelines-worker/templates/_pod.tpl#L117-L121) is created using the ConfigMap from 2.
4. This [volumeMount](https://github.com/DataDog/helm-charts/blob/0e9cc065f09c94cdc54ccdfc3e2f66a9e22620d7/charts/observability-pipelines-worker/templates/_pod.tpl#L84-L86) places the contents of the volume at `/etc/observability-pipelines-worker/` - the configuration we provided creates each key under `data` as a file there.
5. The provided file is accessible at `/etc/observability-pipelines-worker/pipelines.yaml` during runtime.

At this point the only file path the config will be placed is the above, and the only method of providing a config is through the `config` key. Do we think users actually need this supported? Off the top of my head I can't think of anything that would require that.